components-core/src/main/java/com/adyen/checkout/components/core/ComponentAvailableCallback.kt

```diff
@@ -7,6 +7,6 @@
  */
 package com.adyen.checkout.components.core
 
-interface ComponentAvailableCallback {
+fun interface ComponentAvailableCallback {
```

Comment:
Is this needed?

Reply:
Not really, but it's a nice little improvement:
```Kotlin
object : ComponentAvailableCallback {
    override fun onAvailabilityResult(isAvailable: Boolean, paymentMethod: PaymentMethod) {
        updateOutputData(isButtonVisible = isAvailable)

        if (!isAvailable) {
            exceptionChannel.trySend(GooglePayUnavailableException())
        }
    }
}
```
vs
```
{ isAvailable, paymentMethod ->
    updateOutputData(isButtonVisible = isAvailable)

    if (!isAvailable) {
        exceptionChannel.trySend(GooglePayUnavailableException())
    }
}
```
Only downside is if we would ever add a method to the interface then we have to remove the fun and that would break integration if merchants use the lambda way

Reply:
As long as it's not a breaking change currently, I think it's a good idea. I don't see a reason to add a new function there in v5.

Reply:
It's not breaking, it's possible to use both ways now